### DISCUSSION TITLE
Support gender in author metadata?

### DISCUSSION THREAD

**Discussion by sijow** [quarto-dev/quarto-cli#2796] [Feature Requests]
Following https://github.com/quarto-dev/quarto-cli/discussions/1309 quarto uses a neutral form for the author text in French. Maybe the author metadata could also support a "gender", so that translations can use the appropriate male/female/neutral form?

For example

```
---
title: Titre
author:
  name: Jane Doe
  gender: female
lang: fr
---
```

could produce a title block with "Autrice: Jane Doe"  (and "Auteur: Jane Doe" in case of `gender: male`) instead of the current "Auteur·rice: Jane Doe".

I think "auteur·rice" is nice when the gender is not known or when there are several authors of different gender, but when the gender is known it's rather weird... (for readers unfamiliar with French, an English equivalent would be to write "(Wo)man who wrote this: Jane Doe").

For example if I put my name on a document for my students, of course I know my gender and it seems impersonal and strange to write "auteur·rice".

**Comment by mcanouil** [comment]
Even if the field itself is I think a good idea to add.
What you want to do using that field would require to be limited to a list of items (not everyone will recognise themselves in "male", "female" or "neutral"), then this list would need to be matched with the possible translations.

Instead, I would suggest you to simply overwrite the translation either in your document header or in `_quarto.yml`:
```yaml
language: 
  title-block-author-single: "Auteur"
```

**Reply by sijow** [reply]
If it helps, the field could be named `language-gender` instead of `gender` since the point is not to classify people but to let them choose language forms.

The field could be open in which values are accepted. Users could then extend the translation data if they want to use an uncommon value. There could be a recommendation for translations to support at least "male", "female" and "unspecified"[^1] where it makes sense. 

There are of course many more gender "values" in use, in particular for pronouns (see e.g. https://uwm.edu/lgbtrc/support/gender-pronouns/) but maybe for quarto's purpose that's not very relevant? I mean for example I'm not aware of another form than "auteur" and "autrice" or the alternative "auteure". For language forms that are not well known it seems reasonable to let the user define it themselves.

Good point with `title-block-author-single`. But it feels to me this should be handled by the translations, at least for the common cases. It feels cumbersome to have to fix this manually (especially when writing in different languages, having to maintain fixes for each).

[^1]: "unspecified" might be better than "neutral"?

**Reply by mcanouil** [reply]
> It feels cumbersome to have to fix this manually (especially when writing in different languages, having to maintain fixes for each).

Would you not be manually fixing it by having to use/set a `gender` option? How this would be any different than setting the "translation" to whatever you like without any limitations? 🤔

> But it feels to me this should be handled by the translations

Beside the pluralisation, the translations are not scripts. They are simple text that are used at some places depending on `lang` which actually only tell Quarto the file to use, not which text translation.

**Reply by sijow** [reply]
> Would you not be manually fixing it by having to use/set a gender option? How this would be any different than setting the "translation" to whatever you like without any limitations? 🤔

I don't mind defining `gender` to give the system the data it needs to generate the correct output in any language, it makes sense. 

Using `language:\n  title-block-author-single: ...` to override the default behavior feels like a hack, and is not very practical. For example

* I must remember to fiddle with the `language` block and adapt it when the list of author changes.

   "Here's a `_quarto.yml` you can use for this report but don't forget to adapt `title-block-author-single` under `language`!

   "I have a coauthor now, I must set `title-block-author-plural`."

   "A third coauthor of different gender? Don't forget to adapt `title-block-author-plural`!"

* I often write in English, French and German, so I have to find the appropriate words in each language and remember to set the `language` block appropriately depending on the document's language.

* I often generate French and German from the same source file. Having to take care of `title-block-author-XXX` in this context is not making things easier... (Probably something can be done with project profiles in the future quarto 1.2, but that's increasing the whole project's complexity just for this).

All these could be handled properly and automatically by quarto... it just needs this field in author metadata :-)

> Beside the pluralisation, the translations are not scripts.  They are simple text that are used at some places depending on `lang` which actually only tell Quarto the file to use, not which text translation.

Why not handle it the same way as singular/plural? On top of`-single` and `-plural` (which could still be used for unspecified or mixed gender) the translation file could add optionally `-single-m`, `-single-f`, `-plural-m`, `plural-f`, where it makes sense.

**Reply by mcanouil** [reply]
> it just needs this field in author metadata :-)

No, it's not that simple.
Adding the new field won't change anything. A new step in the pre-processing step is required. As I said YAML is not a programming/scripting language.

To be clear and as I said at the beginning, a gender field makes sense to me and why not add some pre-processing along with it, but the expected output and use case have to be clear to ensure a smooth and useful implementation by Quarto team if they agree with it.

**Reply by sijow** [reply]
Ah I meant that the `gender` field is the only thing quarto needs from the user, to properly handle situations with changing lists of authors and different languages. It's true that some logic will be required in quarto to make use of this field, but that's nothing special: many YAML keys require some logic (`title-block-author-plural`, `date`, `execute: cache`, `keep-tex` just to give a few examples).

**Comment by mcanouil** [comment]
FYI @dragonstyle as you opened few issues about the author schema.

**Comment by jawadakhan97** [comment]
Hello, I was looking for a pronoun option as well for a blog I am making. I wanted to add it to the `yaml` in the `.qmd` preceeding the document body.

something like 
```
---
title: "Paradise Lost"
author: "John Milton"
author-pronouns: "he / him"
categories:
  - Epic Poetry
  - Blank Verse
  - Religious Fiction
  - Christian Mythology
---
```
or for an extended `author` section
```
---
title: "Paradise Lost"
author: 
  name: "John Milton"
  pronouns: "he / him"
  gender: "man"
categories:
  - Epic Poetry
  - Blank Verse
  - Religious Fiction
  - Christian Mythology
---
```

Thanks for the work! Really liking `quarto a ton`

**Comment by mcanouil** [comment]
I opened an issue to track this:
- #8476
